### PR TITLE
docs: fix deprecation guide in release notes

### PR DESCRIPTION
**Pull request by majorgreys**
We made some changes in prep for v0.59.0 that require updating the guidance in this release note.

### CODE REVIEW

**Review by brettlangdon — DISMISSED** [review]
I wonder if we should also document `python -Wd` / `PYTHONWARNINGS="default"` to not raise errors but to just log the deprecation warnings?

**Comment by majorgreys** [comment]
@Mergifyio backport 1.0

File: releasenotes/notes/release-1.0-758fc917e40d7eb3.yaml [hunk 1]
```diff
@@ -18,12 +18,17 @@ prelude: >
...
 
   .. note::
-  Before upgrading to v1.0.0, we recommend installing ``ddtrace>=0.59.0,<1.0.0`` and running tests with `warning filters <https://docs.python.org/3.8/library/warnings.html#warning-filter>`_ set to raise ``ddtrace.warnings.DDTraceV1DeprecationWarning`` as errors or set the environment variable ``DD_TRACE_RAISE_DEPRECATIONWARNING``. For example::
+  Before upgrading to v1.0.0, we recommend installing ``ddtrace>=0.59.0,<1.0.0`` and configuring `warning filters <https://docs.python.org/3.8/library/warnings.html#warning-filter>`_ to show warnings for the category ``ddtrace.DDTraceDeprecationWarning``.
 
-    $ DD_TRACE_RAISE_DEPRECATIONWARNING=1 python tests.py
-    $ pytest -W "error::ddtrace.warnings.DDTraceDeprecationWarning" tests.py
-    
-  Once all deprecation warnings are addressed you can safely upgrade to v1.0.0.
+  Users of pytests can add a filter for ``ddtrace.DDTraceDeprecationWarning``::
```
Comment (Kyle-Verhoog) [inline releasenotes/notes/release-1.0-758fc917e40d7eb3.yaml:23]:
```suggestion
  Users of pytest can add a filter for ``ddtrace.DDTraceDeprecationWarning``::
```

**Review by Kyle-Verhoog — COMMENTED** [review]
Just one catch, otherwise lgtm

File: releasenotes/notes/release-1.0-758fc917e40d7eb3.yaml (hunk 1 above)
Comment (mabdinur) [reply releasenotes/notes/release-1.0-758fc917e40d7eb3.yaml:23]: Aside from this LGTM

**Review by brettlangdon — APPROVED** [review]

**Comment by majorgreys** [comment]
image

**Review by mabdinur — APPROVED** [review]

**Pull request merged**